### PR TITLE
allow no counts from dedup

### PR DESCRIPTION
**Pull request by rs-tohkubo**
-deleted

### CODE REVIEW

File: tap_s3_csv/conversion.py [hunk 1]
```diff
@@ -111,6 +111,9 @@ def generate_schema(samples):
     to_return = {}
     counts = count_samples(samples)
 
+    if not counts:
+        return to_return
```
Comment (rs-mkaiser) [inline tap_s3_csv/conversion.py:115]: okay so if generate schema returns here, won't the code relying on the schema then fail? Have you tested it?

**Comment by rs-mkaiser** [comment]
no tests? How do you know this change fixes anything?

**Review by rs-mkaiser — CHANGES_REQUESTED** [review]
need either really convincing answers to my questions that shows that this will fix the problem, or some tests

**Comment by rs-tohkubo** [comment]
Added a test to make sure that when no samples are received after dedup, the counts dictionary won't be populated and added a check to make sure it doesn't break when counts returns an empty dict.

File: tap_s3_csv/conversion.py (hunk 1 above)
Comment (rs-mkaiser) [reply tap_s3_csv/conversion.py:115]: My question is still not addressed

Comment (rs-tohkubo) [reply tap_s3_csv/conversion.py:115]: the case we're trying to handle is when counts is None. The unit test I added handles that case.

Comment (rs-mkaiser) [reply tap_s3_csv/conversion.py:115]: **won't the code relying on the schema then fail? Have you tested it?**

**Comment by rs-tohkubo** [comment]
Upon further investigation, it ended up being a simple version bump of singer docker component (was using an outdated version).

**Pull request closed without merging**